engine/access/rpc/backend/backend_transactions.go

```diff
@@ -240,20 +240,68 @@ func (b *backendTransactions) GetTransactionResult(
 	}
 
 	// derive status of the transaction
-	status, err := b.deriveTransactionStatus(tx, transactionWasExecuted, block)
+	txStatus, err := b.deriveTransactionStatus(tx, transactionWasExecuted, block)
 	if err != nil {
 		return nil, convertStorageError(err)
 	}
 
 	return &access.TransactionResult{
-		Status:       status,
+		Status:       txStatus,
 		StatusCode:   uint(statusCode),
 		Events:       events,
 		ErrorMessage: txError,
 		BlockID:      blockID,
 	}, nil
 }
 
+func (b *backendTransactions) GetTransactionResultByIndex(
+	ctx context.Context,
+	blockID flow.Identifier,
+	index uint32,
+) (*access.TransactionResult, error) {
+	// TODO: Merge flow.TransactionResult and access.TransactionResult so caching doesn't cause a circular dependency
+	block, err := b.blocks.ByID(blockID)
+	if err != nil && !errors.Is(err, storage.ErrNotFound) {
+		return nil, convertStorageError(err)
+	}
```

Comment:
But, what happens if the error is `storage.ErrNotFound`? We still need to fail here? I think you need to remove the second condition in the if statement.